aspnetcore/performance/ObjectPool/ObjectPoolSample8/Program.cs

```diff
@@ -0,0 +1,58 @@
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.ObjectPool;
+using System.Security.Cryptography;
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.TryAddSingleton<ObjectPoolProvider, DefaultObjectPoolProvider>();
+
+builder.Services.TryAddSingleton<ObjectPool<ReusableBuffer>>(serviceProvider =>
+{
+    var provider = serviceProvider.GetRequiredService<ObjectPoolProvider>();
+    var policy = new DefaultPooledObjectPolicy<ReusableBuffer>();
+    return provider.Create(policy);
+});
+
+var app = builder.Build();
+
+app.MapGet("/", () => "Hello World!");
+
+// return the SHA256 hash of a word 
+// https://localhost:7214/SamsonAmaugo
+app.MapGet("/{name}", async (string name, ObjectPool<ReusableBuffer> bufferPool) =>
+{
+
+    var buffer = bufferPool.Get();
+    try
+    {
+        // Set the buffer data to the ASCII values of a word
+        for (int i = 0; i < name.Length; i++)
+        {
+            buffer.Data[i] = (byte)name[i];
+        }
+
+        using (var sha256 = SHA256.Create())
+        {
+            // Calculate the hash of the data
+            var hash = sha256.ComputeHash(buffer.Data);
+            return "Hash: " + BitConverter.ToString(hash).Replace("-", "");
+        }
```

Comment:
```suggestion
        Span<byte> hash = stackalloc byte[32];
        SHA256.HashData(buffer.Data.Slice(name.Length), hash);
        return "Hash: " + Convert.ToHexString(hash);
```
1. Prefer static hash data method https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1850
2. Only want to hash the part of the array that's used
3. Can stackalloc destination for hash calculation
4. There is a convert method that doesn't include dashes: https://learn.microsoft.com/en-us/dotnet/api/system.convert.tohexstring?view=net-7.0

A bunch of clean up. I think that's it 😄

Reply:
@JamesNK Slice doesn't seem to be available for the array byte data type